gmso/core/improper.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+
+from gmso.core.connection import Connection
+from gmso.core.improper_type import ImproperType
+from gmso.exceptions import GMSOError
+
+
+class Improper(Connection):
+    """A 4-partner connection between sites.
+
+    This is a subclass of the gmso.Connection superclass.
+    This class has strictly 3 members in its connection_members.
+    The connection_type in this class corresponds to gmso.ImproperType
+
+    Parameters
+    --------
+    connection_members: list of gmso.Site
+        4 sites of a improper. Central site first, then three
+        sites connected to the central site.
+    connection_type : gmso.ImproperType, optional, default=None
+        ImproperType of this improper.
+    name : str, optional, default=Improper
+        Name of the improper.
+
+    Notes
+    -----
+    Inherits some methods from Connection:
+        __eq__, __repr__, _validate methods
+    Addiitonal _validate methods are presented
+    """
+
+    def __init__(self, connection_members=[], connection_type=None, name="Improper"):
+        connection_members = _validate_four_partners(connection_members)
+        connection_type = _validate_impropertype(connection_type)
+
+        super(Improper, self).__init__(connection_members=connection_members,
+                connection_type=connection_type, name=name)
+
+
+def _validate_four_partners(connection_members):
+    """Ensure 4 partners are involved in Improper"""
+    assert connection_members is not None, "connection_members is not given"
+    if len(connection_members) != 4:
+        raise GMSOError("Trying to create an Improper "
+                "with {} connection members". format(len(connection_members)))
+    return connection_members
```

Comment:
Is there a way to validate that the 4 connection members form an improper instead of a dihedral? This would be done easier with a bond graph, but maybe we can stick a hacky improper validation + test for now

Reply:
Yeah I was wondering that as well. Unless I'm mistaken, we don't have any analogous checks for angles or dihedrals at the moment. It would however seem sensible to enforce the sorts of connectivity requirements that you describe. I didn't add anything because I'm a bit disconnected from the core development effort here and wasn't sure what the thinking was on that front.

Reply:
This validation could be part of another PR. I reccomend raising an issue with a running list of things left to do for full integration of impropers (add them for the ff and xml, write impropers to the various file formats, validate improper members, etc.)

Reply:
Not sure the best place to put it, but we worked on [detecting bond patterns in an earlier PR](https://github.com/mosdef-hub/gmso/pull/140/files)